cloudinit/sources/__init__.py

```diff
@@ -218,6 +221,13 @@ def __init__(self, sys_cfg, distro, paths, ud_proc=None):
         else:
             self.ud_proc = ud_proc
 
+    def _unpickle(self, ci_pkl_version: int) -> None:
+        """Perform deserialization fixes for Paths."""
+        if not hasattr(self, 'vendordata2'):
+            self.vendordata2 = None
+        if not hasattr(self, 'vendordata2_raw'):
+            self.vendordata2_raw = None
```

Comment:
Obviously we need to address this issue so this isn't even a nit, but is it a bit strange that `vendordata2` uses attributes on `DataSource` when it's an OpenStack-specific concept?

Reply:
Yeah, I mistakenly assumed that it had to sit at that same level of abstraction of the parsing of other vendor data. There's been some chatter happening about it here: https://github.com/canonical/cloud-init/pull/777#discussion_r591661062